src/main/kotlin/com/sourcegraph/cody/agent/protocol/util/Rfc3986UriEncoder.kt

```diff
@@ -0,0 +1,15 @@
+package com.sourcegraph.cody.agent.protocol.util
+
+object Rfc3986UriEncoder {
+
+  // todo solve this with library
+  fun encode(uri: String): String {
+    val isWindowsPath = uri.matches("^file:///[A-Z]:/.*".toRegex())
```

Comment:
does it make sense to extract `"^file:///[A-Z]:/.*".toRegex()` to a field? idk what `toRegex` does and if there is any optimization

Reply:
Possibly. From what I've seen, the matcher is statefu but the compiled Regex itself might be held as a static field. But I don't think we should worry about that.